src/NextGen/DonationForm/Blocks/DonationFormBlock/Controllers/BlockRenderController.php

```diff
@@ -44,12 +47,35 @@ public function render(array $attributes)
         $donationForm = DonationForm::find($blockAttributes->formId);
 
         $viewUrl = (new GenerateDonationFormViewRouteUrl())($donationForm->id);
+        $embedId = $blockAttributes->blockId;
+
+        if ($this->shouldDisplayDonationConfirmationReceipt($embedId)) {
+            $receiptId = give_clean($_GET['givewp-receipt-id']);
+
+            $viewUrl = (new GenerateDonationConfirmationReceiptViewRouteUrl())($receiptId);
+        }
 
         /**
          * Note: iframe-resizer uses querySelectorAll so using a data attribute makes the most sense to target.
          * It will also generate a dynamic ID - so when we have multiple embeds on a page there will be no conflict.
          */
-        return "<iframe data-givewp-embed src='$viewUrl'
-                style='width: 1px;min-width: 100%;border: 0;'></iframe>";
+        return "<iframe data-givewp-embed src='$viewUrl' data-givewp-embed-id='$embedId' style='width: 1px;min-width: 100%;border: 0;'></iframe>";
+    }
+
+    /**
+     * @unreleased
+     */
+    protected function shouldDisplayDonationConfirmationReceipt(string $embedId): bool
+    {
+        $routeListener = new RouteListener(
+            'donation-completed',
+            'show-donation-confirmation-receipt'
+        );
+
+        return $routeListener->isValid($_GET, function ($request) use ($embedId) {
+            $isset = isset($request['givewp-embed-id'], $request['givewp-receipt-id']);
+            
+            return $isset && $request['givewp-embed-id'] === $embedId && strlen($request['givewp-receipt-id']) === 32;
```

Comment:
Is there a way around using a magic number here? Can we set that to an Enum or a class constant?

Reply:
Sure, the number 32 is just the standard character length of an md5 hash.  I can add something that makes that more clear.

Reply:
resolved here for now [fd7d6bc](https://github.com/impress-org/givewp-next-gen/pull/113/commits/fd7d6bc273399386061b1aecced148c47e7fe357).